multipass-addressing/src/main/scala/com/precisely/bigdata/sample/spark/MultipassAddressing.scala

```diff
@@ -0,0 +1,181 @@
+ * specific language governing permissions and limitations under the License.
+ */
+
+package com.precisely.bigdata.sample.spark
+
+import com.pb.downloadmanager.api.DownloadManagerBuilder
+import com.pb.downloadmanager.api.downloaders.LocalFilePassthroughDownloader
+import com.pb.downloadmanager.api.downloaders.hadoop.{HDFSDownloader, S3Downloader}
+import com.precisely.addressing.v1.model.{PreferencesBuilder, RequestAddress}
+import com.precisely.bigdata.addressing.spark.api.{AddressingBuilder, AddressingExecutor, RequestInput}
+import org.apache.commons.collections.CollectionUtils
+import org.apache.spark.SparkConf
+import org.apache.spark.sql.expressions.UserDefinedFunction
+import org.apache.spark.sql.functions.{col, lit, map}
+import org.apache.spark.sql.types.{StringType, StructField, StructType}
+import org.apache.spark.sql.{DataFrame, SaveMode, SparkSession}
+
+import java.util.Collections
+
+
+object MultipassAddressing {
+  def main(args: Array[String]): Unit = {
+    val resourcesLocation = args(0)
+    val dataLocation = args(1)
+    val downloadLocation = args(2)
+    val inputAddressPath = args(3)
+    val outputDirectory = args(4)
+
+    val sparkConf = new SparkConf()
+    sparkConf.setIfMissing("spark.master", "local[*]")
+    val session = SparkSession.builder()
+      .config(sparkConf)
+      .getOrCreate()
+
+    // Load the addresses from the csv
+    val addressInputDF = AddressInput.open(session, inputAddressPath)
+
+    val downloadManager = new DownloadManagerBuilder(downloadLocation)
+      .addDownloader(new S3Downloader(session.sparkContext.hadoopConfiguration))
+      .addDownloader(new HDFSDownloader(session.sparkContext.hadoopConfiguration))
+      .addDownloader(new LocalFilePassthroughDownloader())
+      .build()
+
+    // build a AddressingUDF, with custom executor
+    val addressingUdf: UserDefinedFunction = new AddressingBuilder()
+      .withResourcesLocation(resourcesLocation)
+      .withDataLocations(dataLocation)
+      .withDownloadManager(downloadManager)
+      .udfBuilder()
+      .withOutputFields(
+        "location.feature.geometry.coordinates.x as X",
+        "location.feature.geometry.coordinates.y as Y",
+        "address.formattedStreetAddress as formattedStreetAddress",
+        "address.formattedLocationAddress as formattedLocationAddress",
+        "customFields['PRECISION_CODE'] as precisionCode")
+      .withErrorField("error")
+      .forCustomExecutor(new CustomExecutor())
+
+    // call UDF for each row in the address DataFrame, this will result in the dataframe containing a new column for each
+    // of the specified output fields as well as an error column
+    addressInputDF
+      // Adds a new column, represented as a collection comprised of the outputFields and the error field
+      .withColumn("addressing_geocode_result", addressingUdf(map(
+        lit("street"), col("streetName"),
```

Comment:
lets make this addressLines[0]

Reply:
Is there any specific reason to change it from street to addressLines[0]. I just want to understand the change here

Reply:
yes